gpAux/extensions/pxf/src/pxfprotocol.c

```
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+#include "postgres.h"
+#include "fmgr.h"
+#include "access/extprotocol.h"
+
+/* define magic module unless run as a part of test cases */
+#ifndef UNIT_TESTING
+PG_MODULE_MAGIC;
+#endif
+
+PG_FUNCTION_INFO_V1(pxfprotocol_export);
+PG_FUNCTION_INFO_V1(pxfprotocol_import);
+PG_FUNCTION_INFO_V1(pxfprotocol_validate_urls);
+
+Datum pxfprotocol_export(PG_FUNCTION_ARGS);
+Datum pxfprotocol_import(PG_FUNCTION_ARGS);
+Datum pxfprotocol_validate_urls(PG_FUNCTION_ARGS);
+
+/* include explicitly as cdbvars.h is not available when compiling as an extension */
```

This is really fragile. This will silently fail and crash if the GpId struct changes. Why can cdbvars.h not be included here?